### PR TITLE
tests: handle multiple docker versions

### PR DESCRIPTION
**Pull request by crazy-max**
* ~needs https://github.com/moby/buildkit/pull/4736~

### CODE REVIEW

**Comment by crazy-max** [comment]
@tonistiigi As discussed we could also handle builders through `docker context` with specific dind image as follow-up.

**Comment by davhdavh** [comment]
These seem relevant to this pr:
https://github.com/docker/buildx/blob/4327ee73b18949e47eb1c865daa262af095cf974/tests/build.go#L442-L443
https://github.com/docker/buildx/blob/4327ee73b18949e47eb1c865daa262af095cf974/tests/build.go#L576-L577
https://github.com/docker/buildx/blob/4327ee73b18949e47eb1c865daa262af095cf974/tests/build.go#L643-L644

**Comment by crazy-max** [comment]
> These seem relevant to this pr:
> 
> https://github.com/docker/buildx/blob/4327ee73b18949e47eb1c865daa262af095cf974/tests/build.go#L442-L443
> 
> https://github.com/docker/buildx/blob/4327ee73b18949e47eb1c865daa262af095cf974/tests/build.go#L576-L577
> 
> https://github.com/docker/buildx/blob/4327ee73b18949e47eb1c865daa262af095cf974/tests/build.go#L643-L644

WDYM? This PR is about testing multiple dockerd versions. For BuildKit see https://github.com/docker/buildx/pull/2322

File: Dockerfile [hunk 1]
```diff
@@ -5,6 +5,7 @@ ARG XX_VERSION=1.5.0
 
 # for testing
 ARG DOCKER_VERSION=27.4.0-rc.2
+ARG DOCKER_VERSION_ALT_26=26.1.3
```
Comment (crazy-max) [inline Dockerfile:8]: 26.1.3 is the current version shipped on GitHub runners

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**